sources/prairie-dog/src/main.rs

```diff
@@ -0,0 +1,431 @@
+        LOCAL_BOOT_MOUNT_PATH,
+        Some("ext4"),
+        nix::mount::MsFlags::MS_RDONLY | flags,
+        Some(BOOT_MOUNT_SELINUX_CONTEXT),
+    )
+    .context(error::Mount {
+        path: LOCAL_BOOT_MOUNT_PATH,
+    })?;
+
+    info!("Creating bind mount in {}", BOOT_MOUNT_PATH);
+    // Create a bind mount from /local/boot to /boot for consistency with
+    // /var and /opt
+    nix::mount::mount(
+        Some(LOCAL_BOOT_MOUNT_PATH),
+        BOOT_MOUNT_PATH,
+        NONE,
+        nix::mount::MsFlags::MS_BIND | flags,
+        NONE,
+    )
+    .context(error::Mount {
+        path: BOOT_MOUNT_PATH,
+    })?;
+
+    // Make the mount points private so new mount namespaces don't have
+    // access to it. This has to be set as a different call otherwise
+    // the mount syscall returns `EINVAL`
+    make_private(LOCAL_BOOT_MOUNT_PATH)?;
+    make_private(BOOT_MOUNT_PATH)?;
+
+    Ok(())
+}
+
+// Sets the mount point's propagation as private
+fn make_private<S>(path: S) -> Result<()>
+where
+    S: AsRef<str>,
+{
+    let path = path.as_ref();
+    nix::mount::mount(NONE, path, NONE, nix::mount::MsFlags::MS_PRIVATE, NONE)
+        .context(error::SetupMount { path: path })
+}
+
+/// Loads the crash kernel using kexec-tools
+fn load_crash_kernel() -> Result<()> {
+    let kexec_crash_size_path = Path::new(KEXEC_CRASH_SIZE);
+    let kexec_crash_size = fs::read(kexec_crash_size_path).context(error::ReadFile {
+        path: kexec_crash_size_path,
+    })?;
+    let memory_allocated = String::from_utf8_lossy(&kexec_crash_size);
+
+    // We provide a more useful message when no memory was reserved for the crash kernel.
+    // This could happen when no sufficient memory is available in the host.
+    if memory_allocated.trim() == "0" {
+        return error::CrashKernelReservedMemory.fail();
+    }
+
+    let kexec_load_disabled_path = Path::new(KEXEC_LOAD_DISABLED);
+    let kexec_load_disabled_value =
+        fs::read(kexec_load_disabled_path).context(error::ReadFile {
+            path: kexec_load_disabled_path,
+        })?;
+    let kexec_load_disabled = String::from_utf8_lossy(&kexec_load_disabled_value).trim() == "1";
+
+    // We provide a more useful message when `kexec_load_disabled` is set to 1
+    if kexec_load_disabled {
+        return error::KexecLoadDisabled.fail();
+    }
+
+    info!("Loading crash kernel");
+    // Load the panic kernel from `BOOT_MOUNT_PATH`, letting kexec decide which syscall
+    // it should it use (KEXEC_LOAD and KEXEC_FILE_LOAD).
+    // We reuse the cmd line in /proc/cmdline to boot the crash kernel, and we start
+    // a specific systemd service instead of the default target.
+    // Set children-max to prevent OOM errors caused by udev events.
+    command(
+        KEXEC_PATH,
+        &[
+            "-pa",
+            "--reuse-cmdline",
+            "--append",
+            "irqpoll maxcpus=1 reset_devices systemd.unit='kdump.service' udev.children-max=1",
```

Comment:
`irqpoll` isn't one of the recommended flags for arm64

Reply:
I updated the cmdline parameters, and now we are using what AL2 uses, plus the recommended parameters.